pkg/monitoring/rules/recordingrules/operator.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2024 The KubeVirt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recordingrules
+
+import (
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	"github.com/machadovilaca/operator-observability/pkg/operatorrules"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	abnormalLabelKey = "reason"
+)
+
+var operatorRecordingRules = []operatorrules.RecordingRule{
+	{
+		MetricsOpts: operatormetrics.MetricOpts{
+			Name: "kubevirt_memory_working_set_to_requested_diff_bytes",
+			Help: "The diff between the pod with highest memory working set and its requested memory for each container, virt-controller, virt-handler, virt-api and virt-operator.",
+			ConstLabels: map[string]string{
+				abnormalLabelKey: "WorkingSetMemoryExceeded",
```

Comment:
_reposting my comment since the other discussion has been marked as resolved_

you could do differently: have a single `kubevirt_current_memory_to_requests_delta_bytes` + a "source" label identifying if the diff is from the working_set or RSS memory.
The reason for having a single metric is that "or" joins are harder to get right. 

```
kubevirt_memory_working_set_to_requested_diff_bytes or kubevirt_memory_rss_to_requested_diff_bytes
````

vs.

````
topk by(container) (1, kubevirt_current_memory_to_requests_delta_bytes > 0)
````

Reply:
but then I will need to use the "source" label inside the Expr no?

Reply:
From a PromQL/Prometheus standpoint, the idiomatic way is to use the same metric name with labels to differentiate the source. It allows things like "tell me the top 10 values irrespective of the source" which wouldn't work if you have 2 metric names.

Reply:
Also if you want to go the route of 2 different metrics, the "source" label is unnecessary since the metric names tell you the same.

Reply:
Both methods are fine by me.

Reply:
will appreciate if we can decide on the method, so we can go forward with this pr @simonpasquier @sradco thanks!

Reply:
@avlitman Please update the metric to:

`kubevirt_memory_delta_from_requested_bytes` with a `source` label (`memory_rss`, `memory_working_set`) and update the recording rule accordingly. Thanks.

Reply:
fixed